app/src/main/java/com/babylon/wallet/android/presentation/account/AccountViewModel.kt

```diff
@@ -48,10 +50,15 @@ class AccountViewModel @Inject constructor(
 
     init {
         viewModelScope.launch {
-            getProfileUseCase.accountOnCurrentNetworkWithAddress(accountAddress).collectLatest { account ->
-                account?.let {
+            getProfileUseCase.accountOnCurrentNetworkWithAddress(accountAddress).collectLatest { accountFromAddress ->
+                accountFromAddress?.let { account ->
+                    val dashboardUrl = NetworkId.from(account.networkID).dashboardUrl()
+                    val historyDashboardUrl = "$dashboardUrl/account/$accountAddress/recent-transactions"
                     _state.update { state ->
-                        state.copy(accountWithAssets = AccountWithAssets(account = account, assets = null))
+                        state.copy(
+                            accountWithAssets = AccountWithAssets(account = account, assets = null),
+                            historyDashboardUrl = historyDashboardUrl
```

Comment:
This works fine, but I propose a suggestion that works in cases like these. In this example what we need as an input in order to determine the URL exists already in the `State()` class. 

So we could create a backing field in the State class like:
```
val historyDashboardUrl: Uri?
    get() = accountWithAssets?.account?.let { account ->
        val dashboardUrl = NetworkId.from(account.networkID).dashboardUrl()
        return Uri.parse("$dashboardUrl/account/${account.address}/recent-transactions")
    }
```

So there is no need to add more explicit code into the viewmodel itself since the state can derive the field by the rest of the fields. (I just wrote the code, it may not compile)

Reply:
> opose a suggestion that works in cases like these. In this example what we need as an input in order to determine the URL exists already in the `State()` class.

Yes 👍